dev-games/recastnavigation/recastnavigation-1.5.1_p20230102.ebuild

```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Navigation mesh construction toolset for games"
+HOMEPAGE="https://github.com/recastnavigation/recastnavigation"
+MY_COMMIT="405cc095ab3a2df976a298421974a2af83843baf"
+SRC_URI="https://github.com/recastnavigation/recastnavigation/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-${MY_COMMIT}"
+
+LICENSE="ZLIB"
+SLOT="0/1.5.1"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+DEPEND="test? ( dev-cpp/catch:0 )"
+
+PATCHES=(
+	"${FILESDIR}/${P}-catch.patch"
+)
+
+src_prepare() {
+	rm Tests/Contrib/Catch/* || die
+	use test && echo "#include <catch2/catch_all.hpp>" > Tests/Contrib/Catch/catch_amalgamated.hpp
```

Comment:
we usually consider conditional patch as a bad approach, is there any problem to make an unconditional patch? besides, it would be great to push the patch to upstream (to have clean ebuild in next version bump, less maintenance burden)

also, I'd personally prefer a patch file here, call external command (echo) may encounter a failure, so should check return value and a "|| die" is necessary..

Reply:
> is there any problem to make an unconditional patch? 

I intended this to be a kind of comment: to make it more clear that this is relevant only for the tests (even though the file touched has Tests in it)

I didn't go with patch here because:
1. the cleanest way is probably to replace `#include "catch_amalgamated.hpp"` in the several cpp files with including `catch2/catch_all.hpp` directly, but for that I'd need to edit more files than this, therefore it's more fragile if a set of includes around this one changes
2. replacing the whole bundled Catch2 copy via a patch would require a huge patch
3. after the rm, adding the one line via a patch which takes ---/dev/null would be weird

Anyway, if upstream accepts the changes, all this would be moot. I'll try to send it to them soon. For now removed condition and added die

Reply:
https://github.com/recastnavigation/recastnavigation/pull/600 but this is more complicated patch than what's done here